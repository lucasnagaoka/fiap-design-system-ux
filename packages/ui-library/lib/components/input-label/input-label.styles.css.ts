import { style } from '@vanilla-extract/css';

const inputLabel = style({
  display: 'flex',
  flexDirection: 'column',
  width: '100%',
});

export { inputLabel };
