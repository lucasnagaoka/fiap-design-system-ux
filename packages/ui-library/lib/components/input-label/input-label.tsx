import { useId, type JSX } from 'react';
import type { InputLabelProps } from './input-label.types';
import { Label } from '../label/label';
import { Input } from '../input/input';

export function InputLabel({
  label,
  variant = 'primary',
  labelProps,
  inputProps,
}: InputLabelProps): JSX.Element {
  const id = useId();

  return (
    <>
      {label && (
        <Label htmlFor={`${id}-${label}`} {...labelProps}>
          {label}
        </Label>
      )}
      <Input id={`${id}-${label}`} variant={variant} {...inputProps} />
    </>
  );
}

InputLabel.displayName = 'InputLabel';
