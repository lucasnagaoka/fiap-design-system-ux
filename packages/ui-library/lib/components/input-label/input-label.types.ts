import type { InputProps } from '../input/input.types';
import type { LabelProps } from '../label/label.types';

export type InputLabelProps = {
  label: string;
  variant?: 'primary' | 'secondary';
  labelProps?: LabelProps;
  inputProps?: InputProps;
};
