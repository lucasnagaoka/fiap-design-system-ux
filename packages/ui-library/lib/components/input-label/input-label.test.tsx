import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { InputLabel } from './input-label';

describe('InputLabel component', () => {
  it('Should render the label and input when a label is provided', () => {
    render(<InputLabel label='Nome' inputProps={{ variant: 'primary' }} />);
    const label = screen.getByText('Nome');
    const input = screen.getByRole('textbox');
    expect(label).toBeInTheDocument();
    expect(input).toBeInTheDocument();
  });

  it('Should correctly associate the label with the input using htmlFor and id', () => {
    render(<InputLabel label='Email' />);
    const label = screen.getByText('Email');
    const input = screen.getByRole('textbox');

    expect(label).toHaveAttribute('for', input.getAttribute('id'));
  });

  it('Should apply the input props correctly', () => {
    render(
      <InputLabel label='Senha' inputProps={{ placeholder: 'Password' }} />
    );
    const input = screen.getByPlaceholderText('Password');
    expect(input).toBeInTheDocument();
  });

  it('Should take an input-label snapshot', () => {
    const { asFragment } = render(<InputLabel label='Email' />);

    expect(asFragment).toMatchSnapshot();
  });
});
