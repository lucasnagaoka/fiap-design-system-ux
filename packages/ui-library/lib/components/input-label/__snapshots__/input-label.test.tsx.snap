// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputLabel component > Should take an input-label snapshot 1`] = `[Function]`;
